vault/logical_system.go

```diff
@@ -1159,6 +1192,108 @@ func (b *SystemBackend) handlePluginReloadUpdate(req *logical.Request, d *framew
 	return nil, nil
 }
 
+// handleRequestPassthroughHeaderRead reads the config
+func (b *SystemBackend) handleRequestPassthroughHeaderList(req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	var currHeaders []string
+	currHeadersRaw := b.Core.passthroughRequestHeaders.Load()
+	if currHeadersRaw == nil {
+		return nil, nil
+	} else {
+		currHeaders = currHeadersRaw.([]string)
+	}
+
+	// Check for a nil slice to have been stored
+	if currHeaders == nil {
+		return nil, nil
+	}
+
+	return logical.ListResponse(currHeaders), nil
+}
+
+// handleRequestPassthroughHeaderUpdateDelete creates or overwrites a header to include, or deletes
+func (b *SystemBackend) handleRequestPassthroughHeaderUpdateDelete(delete bool) func(*logical.Request, *framework.FieldData) (*logical.Response, error) {
+	return func(req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+		header := d.Get("header").(string)
+		if header == "" {
+			return logical.ErrorResponse("missing header name"), nil
+		}
+
+		header = textproto.CanonicalMIMEHeaderKey(header)
+
+		b.Core.passthroughRequestHeadersLock.Lock()
+		defer b.Core.passthroughRequestHeadersLock.Unlock()
+
+		var currHeaders []string
+		currHeadersRaw := b.Core.passthroughRequestHeaders.Load()
+		if currHeadersRaw == nil {
+			if delete {
+				return nil, nil
+			}
+		} else {
+			currHeaders = currHeadersRaw.([]string)
+		}
+
+		// Check for a nil or empty slice to have been stored
+		if len(currHeaders) == 0 {
+			if delete {
+				return nil, nil
+			}
+		}
+
+		var found bool
+		for _, v := range currHeaders {
+			if v == header {
+				found = true
+				break
+			}
+		}
+
+		var persist bool
+		var newHeaders []string
+
+		switch {
+		case delete && found:
+			persist = true
+			newHeaders = make([]string, 0, len(currHeaders)-1)
+			for _, v := range currHeaders {
+				if v != header {
+					newHeaders = append(newHeaders, v)
+				}
+			}
+
+		case !delete && !found:
+			persist = true
+			newHeaders = make([]string, 0, len(currHeaders)+1)
```

Comment:
Doesn't the append call automatically extend the underlying array for the slice? Can we skip the `make` here?